experimental/displaynames/src/provider.rs

```diff
@@ -0,0 +1,36 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+// Provider structs must be stable
+#![allow(clippy::exhaustive_structs, clippy::exhaustive_enums)]
+#![warn(unused_imports)]
+//! Data provider struct definitions for this ICU4X component.
+//!
+//! Read more about data providers: [`icu_provider`]
+
+use icu_provider::prelude::*;
+use tinystr::TinyAsciiStr;
+use zerovec::ZeroMap;
+
+// We use raw TinyAsciiStrs for map keys, as we then don't have to
+// validate them as subtags on deserialization. Map lookup can be
+// done even if they are not valid tags (an invalid key will just
+// become inaccessible).
+type UnvalidatedRegion = TinyAsciiStr<14>;
```

Comment:
Should be 3, and drop the -alt-variant ones

See #1225

Reply:
I have changed this to 3. But the serde struct for CLDR json files still has TinyAsciiStr<14>. I am excluding "alt-variant" while transforming to provider data struct. 
I think this should work, but I considered few other options too:

1) Using serde(skip_serialization_if) in source data struct to remove "alt-variant". But I read that this is problematic.
2) Removing the "alt-variant" from CLDR territories.json files manually. 

Let me know what you think.

Reply:
The cldr_serde version can have a String, and then you convert it to a TinyAsciiStr in the transform function, skipping entries that don't fit.

Reply:
Done!